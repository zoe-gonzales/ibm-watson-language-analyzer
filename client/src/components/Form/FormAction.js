import { useState } from 'react';

const formAction = cb => {
    const [text, setText] = useState('');
    const [method, setMethod] = useState('');

    const handleTextInputChange = e => {
        e.persist();
        const { name, value } = e.target;
        setText(text => ({text, [name]: value}));
    }

    const handleSelect = e => {
        e.persist();
        const { name, value } = e.target;
        setMethod(method => ({method, [name]: value}));
    }

    const handleFormSubmit = e => {
        if (e) e.preventDefault();
        cb();
        setText('');
        setMethod('');
    }

    return (
        text,
        method,
        handleTextInputChange,
        handleSelect,
        handleFormSubmit
    )
}

export default formAction;