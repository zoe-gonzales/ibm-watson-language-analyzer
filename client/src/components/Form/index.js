import React from 'react';
import TextInput from '../TextInput';
import DropDown from '../DropDownSelect';
import Button from '../SubmitButton';
import FormAction from './FormAction';
// import Home from '../../pages/Home';
import API from '../../utils/API';

const Form = () => {
    const { inputs, handleInputChange, handleFormSubmit } = FormAction(() => {
        // make request to API here
        // <Home apiResults={res} />
        let req;
        if (inputs.method === 'Keywords') {
            req = API.getKeywords(inputs.text);
        } else if (inputs.method === 'Categories') {
            req = API.getCategories(inputs.text);
        } else if (inputs.method === 'Emotions') {
            req = API.getEmotions(inputs.text);
        }
        req
        .then(res => {
            console.log(res);
        })
        .catch(err => {
            console.log(err);
        });
    });

    const options = [
        {
            "name": "Keywords",
            "id": 1
        },
        {
            "name": "Categories",
            "id": 2
        },
        {
            "name": "Emotions",
            "id": 3
        }
    ];

    return (
        <form>
            <div className="row">
                <TextInput
                    name="text"
                    placeholder="Enter text here..."
                    value={inputs.text}
                    onChange={handleInputChange} />
            </div>
            <div className="row">
                <DropDown
                    name="method"
                    options={options}
                    value={inputs.method}
                    onChange={handleInputChange} />
                <Button onClick={handleFormSubmit} />
            </div>
        </form>
    )
}

export default Form;