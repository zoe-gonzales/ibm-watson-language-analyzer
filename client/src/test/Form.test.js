import { shallow } from 'enzyme';
import Form from '../components/Form';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Form component', () => {
    it('Form: renders correctly', () => {
        const component = renderer
        .create(<Form />)
        .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('componentWillMount: should set the initialSearchValue to state', () => {
      const wrapper = shallow(<Form initialSearchValue={''} initialRequestType={''} />);
      expect(wrapper.instance().state.textToSearch).toBe('');
      expect(wrapper.instance().state.requestType).toBe('');
    });
    it('handleInputChange: should set state to value in input field', () => {
        const wrapper = shallow(<Form initialSearchValue={'To be or not to be'} initialRequestType={'keywords'}/>);
        const instance = wrapper.instance();
        expect(instance.state.textToSearch).toBe('');
        expect(instance.state.requestType).toBe('');
        instance.handleInputChange();
        expect(instance.state.textToSearch).toBe('To be or not to be');
        expect(instance.state.requestType).toBe('keywords');
    });
    it('handleSubmit: should clear state after submit is made', () => {
        const wrapper = shallow(<Form initialSearchValue={'To be or not to be'} initialRequestType={'keywords'}/>);
        const instance = wrapper.instance();
        expect(instance.state.textToSearch).toBe('To be or not to be');
        expect(instance.state.requestType).toBe('keywords');
        instance.handleSubmit();
        expect(instance.state.textToSearch).toBe('');
        expect(instance.state.requestType).toBe('');
    });
  });