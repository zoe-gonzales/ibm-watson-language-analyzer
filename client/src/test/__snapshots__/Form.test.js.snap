// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Form component correctly 1`] = `
<form>
  <div
    className="row"
  >
    <div>
      <textarea
        className="materialize-textarea"
        id="text"
        onChange={[Function]}
      />
      <label
        for="text"
      >
        Enter text here...
      </label>
    </div>
  </div>
  <div
    className="row"
  >
    <div>
      <select
        id="method"
        onChange={[Function]}
      >
        <option
          value="Keywords"
        >
          Keywords
        </option>
        <option
          value="Categories"
        >
          Categories
        </option>
        <option
          value="Emotions"
        >
          Emotions
        </option>
      </select>
      <label
        for="method"
      >
        Select...
      </label>
    </div>
    <button
      className="btn waves-effect waves-light"
      name="action"
      onClick={[Function]}
      type="submit"
    >
      <i
        className="material-icons right"
      >
        send
      </i>
    </button>
  </div>
</form>
`;
